packages/meditrak-server/src/routes/GETSurveyGroups/assertSurveyGroupsPermissions.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import groupBy from 'lodash.groupby';
+import keyBy from 'lodash.keyby';
+import flattenDeep from 'lodash.flattendeep';
+import { hasBESAdminAccess } from '../../permissions';
+
+export const filterSurveyGroupsByPermissions = async (accessPolicy, models, surveyGroups) => {
+  if (hasBESAdminAccess(accessPolicy)) {
+    return surveyGroups;
+  }
+
+  const surveyGroupIds = surveyGroups.map(sg => sg.id);
+  const surveys = await models.survey.findManyByColumn('survey_group_id', surveyGroupIds);
+  const allSurveyCountryIds = flattenDeep(surveys.map(s => s.country_ids));
+  const allCountries = await models.country.findManyById(allSurveyCountryIds);
+  const countryById = keyBy(allCountries, 'id');
+  const allPermissionGroupIds = surveys.map(s => s.permission_group_id);
+  const allPermissionGroups = await models.permissionGroup.findManyById(allPermissionGroupIds);
+  const permissionGroupById = keyBy(allPermissionGroups, 'id');
```

Comment:
Some of this feels like it should be common across this and the survey permissions checking